source/neuropod/backends/torchscript/test/torchscript_test_utils.hh

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2020 The Neuropod Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include "gtest/gtest.h"
+#include "neuropod/neuropod.hh"
+#include "neuropod/backends/torchscript/torch_tensor.hh"
+
+#include <string>
+#include <vector>
+
+#include <stdlib.h>
+
+void test_torchscript_dict_with_union_value_type_model(neuropod::Neuropod &neuropod)
+{
+#if CAFFE2_NIGHTLY_VERSION >= 20220127
+    // Tests a model that has Dict[str, Union[List[str], torch.Tensor]] as model input type
+    std::vector<int64_t> str_shape = {3};
+    std::vector<int64_t> num_shape = {3, 1};
+    const std::vector<int64_t> expected_output_shape = {3, 2};
+    const std::vector<std::string> a_data = {"a", "b", "c"};
+    const float b_data[] = {1, 2, 3};
+    const float target[] = {1, 1, 2, 2, 0, 3};
+
+    auto a_ten = neuropod.allocate_tensor<std::string>(str_shape);
+    a_ten->copy_from(a_data);
+    auto b_ten = neuropod.allocate_tensor<float>(num_shape);
+    b_ten->copy_from(b_data, 3);
+
+    neuropod::NeuropodValueMap input_data;
+    input_data["a"] = a_ten;
+    input_data["b"] = b_ten;
+
+    const auto output_data = neuropod.infer(input_data);
+
+    const std::vector<float> output_vec = output_data->at("c")->as_typed_tensor<float>()->get_data_as_vector();
+
+    const std::vector<int64_t> out_shape = output_data->at("c")->as_tensor()->get_dims();
+
+    EXPECT_EQ(output_vec.size(), 6);
+    EXPECT_TRUE(std::equal(output_vec.begin(), output_vec.end(), target));
+    EXPECT_TRUE(out_shape == expected_output_shape);
+#endif
+}
+
+void test_torchscript_dict_with_union_value_type_model(const std::string &neuropod_path)
```

Comment:
Move the contents of this function directly into the `TEST` block that references it (since it is likely not going to be used anywhere else)